config/rules.neon

```diff
@@ -62,6 +62,18 @@ services:
                 - 'mysqli_query#1'
                 - 'mysqli_execute_query#1'
 
+    -
+        class: staabm\PHPStanDba\Rules\SyntaxErrorInQueryAssemblerRule
+        tags: [phpstan.rules.rule]
+        arguments:
+            classMethods:
+                - 'staabm\PHPStanDba\Tests\Fixture\Connection::assembleNoArrays'
+                - 'staabm\PHPStanDba\Tests\Fixture\Connection::assembleOneArray#1'
+                - 'staabm\PHPStanDba\Tests\Fixture\Connection::assembleTwoArrays#1,2'
```

Comment:
rule definition only meant for testing, should be declared in the `tests/rules/config/`-files.

in the `config/rules.neon` we should only have namespaces to real world definitions and functions.

Reply:
Hmm, I tried putting them in `tests/*/config/phpstan.neon.dist`, but then the phpunit tests all failed. Adding the services via `getAdditionalConfigFiles()` in the test classes gave me an error, e.g.
> Multiple services of type staabm\PHPStanDba\Rules\SyntaxErrorInQueryAssemblerRule found

Here's my attempt:

https://github.com/hemberger/phpstan-dba/commit/7b9c9d4e5b314e5a1f8d3b191b4fda0ce012b25b

Any idea what's wrong with that? I should note that all the existing test fixtures were already in `config/rules.neon`.

Reply:
> I should note that all the existing test fixtures were already in config/rules.neon.

which ones do you mean exactly?

Reply:
The following are already in `config/rules.neon`.
```
 - 'staabm\PHPStanDba\Tests\Fixture\Connection::preparedQuery'
 - 'staabm\PHPStanDba\Tests\Fixture\PreparedStatement::__construct'
 - 'staabm\PHPStanDba\Tests\Fixture\BaseQueryClass::doQuery'
 - 'staabm\PHPStanDba\Tests\Fixture\BaseQueryClass::doQuery#0'
```

Reply:
We will clean this up separately